### DISCUSSION TITLE
Newbie on foreign key relations

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#8587] [Questions] [answered]
New to relational databases and aware that Im potentially thinking about how to use the power of linking tables in the wrong way.

Table1: Teams
TeamID, TeamName

Table2: Players
PlayerID, PlayerName

Table3: Roles
TeamID (ref to Teams.TeamID), PlayerID (ref to Players.PlayerID)

Query
```
const { data,count,error }=await supabase
.from('Roles')
.select('Players(PlayerName),Teams(TeamName)')
.eq('PlayerID','123')
.limit(10)

```
This returns up to 10 Teams that 123 is a member of, but also obviously 10 times the name of 123.

How can I limit the "Players(PlayerName)" part to 1 but the Teams(TeamName) part to 10 in the select?
What is the format for setting a count on the Teams(TeamName) part solely, and to sort the Teams(TeamName) results?

**Comment by steve-chavez** [comment, marked as answer]
> How can I limit the "Players(PlayerName)" part to 1 but the Teams(TeamName) part to 10 in the select?

If `TeamID` and `PlayerID` are also included in the PK of "Roles", then you could do:

```js
const { data,count,error }=await supabase
.from('Players')
.select('PlayerName,Teams(TeamName)')
.eq('PlayerID','123')
```

Basically it will be detected that Players-Teams are on a many-to-many relationship and the request will give you all the teams that a player belongs to.

**Reply by ghost** [reply]
Thanks. That works as long as TeamID and PlayerID are PKs and thus unique in table Roles, but they cant be since a Player can be a member of multiple Teams. I understand Im missing something here though.

**Reply by steve-chavez** [reply]
>  TeamID and PlayerID are PKs and thus unique in table Roles

I mean that both should be a [composite key](https://en.wikipedia.org/wiki/Composite_key) of Roles.

> since a Player can be a member of multiple Teams

@sharkstate This condition is still true with a composite key and additionally you'd enforce that a player can be a member of a team only once(no multiple affiliations to a team).

**Reply by ghost** [reply]
Thanks for sticking with me. I got this rolling now. Is it possible to add sort and count specifically on the Teams result in above? (from what Ive seen this might not be possible currently, but need views work-arounds). Thanks again.

**Reply by steve-chavez** [reply]
You mean order by a count on `Teams(count)` right? Yeah, that's not possible directly for now but you can create a view or function that does the count.

**Reply by ghost** [reply]
Thanks I ment getting the count of the Teams result:

.from('Players')
.select('PlayerName,Teams(TeamName)',{ count: 'exact' })

Will get me a count of 1 (eg the PlayerName), and cant figure out where to put the { count: 'exact' } to get the count of the resulting Teams.

**Reply by steve-chavez** [reply]
`count: exact` only counts the table in the `from`. A couple of options here:

You could do the request backwards

```
.from('Teams')
.select('TeamName,Players(PlayerName)',{ count: 'exact' })
```

Or do

```
.from('Players')
.select('PlayerName,Teams(count)')
```

Which works but gives you the count in an array of length 1(it's a hack for now since aggregates are not officially supported yet)

**Reply by ghost** [reply]
Got it. Thanks!

**Reply by steve-chavez** [reply]
Forgot about this option. You can order on a foreign table `count` with a computed column: https://github.com/supabase/supabase/discussions/7875
